File: Solutions/Java/Convert_an_array_to_reduced_form.java
class Solution {
    void convert(int[] arr, int n) {
        // code here
        HashMap<Integer,Integer> s = new HashMap<Integer,Integer>();
        int arr1[] = arr.clone();
        Arrays.sort(arr1);
        for (int i=0; i<n; i++) {
            s.put(arr1[i], i);
        }
        for (int i=0; i<n; i++) {
            arr[i]=s.get(arr[i]);
        }
    }
}

File: Solutions/Python/Convert_an_array_to_reduced_form.py
class Solution:
	def convert(self,arr, n):
		# code here
		dct = {}
        a = sorted(arr)
        for i in range(len(a)):
            dct[a[i]] = i
        for i in range(n):
            arr[i] = dct[arr[i]]
        return arr
